### PR TITLE
Dashboard v1 UI rewrite done

### PR DESCRIPTION
**Pull request by avdaredevil**
#### Targets issue #2359

Before | After
--- | ---
Before design for Kubeflow Dashboard | AP's design for Kubeflow Dashboard

Iframe view |
--- |
image

### Responsive
Responsive Menu | Responsive Page with Hamburger
--- | ---
image | image

## TODOs
- Most UI work for now is done _(will confirm with UX folk)_
- Setup a backend to pass data to the dash

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2361)

### CODE REVIEW

**Comment by jlewi** [comment]
@avdaredevil This is great. It would be good to demo this at one of the community meetings once its merged.

**Comment by avdaredevil** [comment]
@jlewi Thanks! Sounds good to me

**Comment by kunmingg** [comment]
/lgtm
/approve
Thanks!

**Comment by abhi-g** [comment]
🎉🎉🎉

File: components/centraldashboard/frontend/index/components.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<base href="https://cdn.rawgit.com/download/polymer-cdn/2.6.0/lib/">
```
Comment (yebrahim) [inline components/centraldashboard/frontend/index/components.html]: We should move away from using the CDN.
Separately, rawgit is going away, we shouldn't rely on it even in the interim.

File: components/centraldashboard/frontend/index/main-page.html [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                    {text: 'Deploy a solution', icon: 'donut-large'},
+                    {text: 'Compare run output', icon: 'donut-large'},
+                    {text: 'Share solution on AI Hub', icon: 'donut-large'},
+                ]},
+                sidebarItemIndex: {type: Number, value: 0},
+                primaryViewIndex: {type: Number, value: 0},
+                homeOrIframeViewIndex: {type: Number, value: 0},
+                url: {type: String, value: ""},
+            }}
+            static get observers() {return [
```
Comment (yebrahim) [inline components/centraldashboard/frontend/index/main-page.html]: cleanup?

**Comment by kunmingg** [comment]
@yebrahim 
Thanks!
Can we select framework in follow up PRs, especially when backend parts are involved?

**Comment by yebrahim** [comment]
My comment was about the use of Polymer, I'm not aware of the backend parts.

**Comment by avdaredevil** [comment]
@yebrahim Let's continue this conversation here: #2390

**Comment by sarahmaddox** [comment]
Hallo @avdaredevil and @abhi-g I'd like to discuss the question of iframing the docs into the Kubeflow UI. I'd prefer not to iframe the doc site into the UI, as I think the experience is on the whole confusing and aesthetically unappealing. The doc website is designed as a standalone site. When iframed into the Kubeflow UI, we have double (and slightly different) branding and doubled/trebled navigation elements. 

It may make sense for the Kubeflow webapps all to be iframed into the UI (I don't want to discuss that issue) but I think it makes a better experience for the docs to be left as a separate website, with links going from the various UI elements to the relevant doc page for contextual help.

People are used to having a separate doc site. If the docs are integrated, we need a fully contextual doc config which pops up exactly the right content at each phase of the UI.

**Pull request merged**